functions/vshn-postgres-func/alerting.go

```diff
@@ -0,0 +1,194 @@
+package vshnpostgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/go-logr/logr"
+	alertmanagerv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	"github.com/vshn/appcat-comp-functions/runtime"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AddUserAlerting adds user alerting to the PostgreSQL instance.
+func AddUserAlerting(ctx context.Context, log logr.Logger, iof *runtime.Runtime, comp *vshnv1.VSHNPostgreSQL) (*vshnv1.VSHNPostgreSQL, error) {
+
+	log.Info("Check if alerting references are set")
+
+	log.V(1).Info("Tranfsorming", "obj", iof)
+
+	err := runtime.AddToScheme(alertmanagerv1alpha1.SchemeBuilder)
+	if err != nil {
+		return comp, err
+	}
+
+	monitoringSpec := comp.Spec.Parameters.Monitoring
+
+	if monitoringSpec.AlertmanagerConfigRef != "" {
+
+		if monitoringSpec.AlertmanagerConfigSecretRef == "" {
+			log.Info("Found AlertmanagerConfigRef but no AlertmanagerConfigSecretRef")
+			return comp, fmt.Errorf("found AlertmanagerConfigRef but no AlertmanagerConfigSecretRef, please specify as well")
+		}
+
+		refName := comp.Spec.Parameters.Monitoring.AlertmanagerConfigRef
+		log.Info("Found an AlertmanagerConfigRef, deploying...", "refName", refName)
+
+		err := deployAlertmanagerFromRef(comp, iof)
+		if err != nil {
+			return comp, err
+		}
+
+	}
+
+	if monitoringSpec.AlertmanagerConfigSpecTemplate != nil {
+
+		if monitoringSpec.AlertmanagerConfigSecretRef == "" {
+			log.Info("Found AlertmanagerConfigTemplate but no AlertmanagerConfigSecretRef")
+			return comp, fmt.Errorf("found AlertmanagerConfigTemplate but no AlertmanagerConfigSecretRef, please specify as well")
+		}
+
+		log.Info("Found an AlertmanagerConfigTemplate, deploying...")
+
+		err := deployAlertmanagerFromTemplate(comp, iof)
+		if err != nil {
+			return comp, err
+		}
+	}
```

Comment:
Nitpick: If both `Ref` and `Template` are specified it will first create a config from the ref and then overwrite it using the template.  That's not really a problem, but IMO could be a bit cleaner and the logs will be confusing.